scipy/stats/_quantile.py

```diff
@@ -383,3 +384,35 @@ def _quantile_bc(y, p, n, method, xp):
     elif method == '_nearest':
         k = xp.round(ij)
     return xp.take_along_axis(y, xp.astype(k, xp.int64), axis=-1)
+
+
+
+@xp_capabilities(skip_backends=[("dask.array", "No take_along_axis yet.")])
+def _xp_searchsorted(x, y, *, side='left', xp=None):
+    # Vectorize np.searchsorted. Assumes search is along last axis, which is always
+    # preserved in the output. Does not support zero-length `x`. For side='right',
+    # NaNs in `y` are inserted to the left, in contrast with np.searchsorted.
+    xp = array_namespace(x, y) if xp is None else xp
+    x, y = _broadcast_arrays((x, y), axis=-1, xp=xp)
+
+    a = xp.full(y.shape, 0)
```

Comment:
If `x` is allocated on non-CPU device (e.g. PyTorch on MPS device) the subsequent call to `a + n_nans` would fail:

```suggestion
    a = xp.full(y.shape, 0, device=xp_device(x))
```

Reply:
yes we typically test for this for functions in array-api-extra with https://github.com/data-apis/array-api-extra/blob/95291456b7cbbe1adaad6e4cf56f284ff2b3441f/tests/conftest.py#L219-L234, e.g. https://github.com/data-apis/array-api-extra/blob/95291456b7cbbe1adaad6e4cf56f284ff2b3441f/tests/test_funcs.py#L357-L359

Reply:
Yes, I know,, but i have actually been ignoring that in all my translations lately. It's easier for me to come back and do a whole bunch of corrections like this at once than to keep it in mind every time I create an array. But I can make the adjustment in this PR.

Reply:
Oh wait we need to re-run CI anyway.